### PR TITLE
Implement facility management UI 424

### PR DESCRIPTION
**Pull request by corypride**
## Description of the change

Build out the front-end capabilities that allow an administrator to see a list of and manage facilities.

- **Related issues**: Link to related issue(s) this PR addresses, if any (use phrases like "fixes #1234" or "closes #1234").
Will Fix: #424 
## Screenshot(s)

image

### CODE REVIEW

File: backend/src/handlers/auth.go [hunk 1]
```diff
@@ -257,7 +257,8 @@ func (srv *Server) handleResetPassword(w http.ResponseWriter, r *http.Request, l
 		return newDatabaseServiceError(err)
 	}
 	if claims.UserID == 1 && claims.Role == "admin" && form.FacilityName != "" {
-		if _, err := srv.Db.UpdateFacility(form.FacilityName, 1); err != nil {
+		facility := models.Facility {Name: form.FacilityName,Timezone: "America / Chicago"}
```
Comment (PThorpe92) [inline backend/src/handlers/auth.go:260]: Timezones/tz need to be formatted without a space e.g `America/Chicago`

File: backend/src/handlers/facilities_handler.go [hunk 2]
```diff
@@ -16,14 +16,28 @@ func (srv *Server) registerFacilitiesRoutes() {
...
 		return newDatabaseServiceError(err)
 	}
-	return writeJsonResponse(w, http.StatusOK, facilities)
+	last := srv.calculateLast(total, perPage)
+	paginationData := models.PaginationMeta{
+		PerPage:     perPage,
+		LastPage:    int(last),
+		CurrentPage: page,
+		Total:       total,
+	}
```
Comment (calisio) [inline backend/src/handlers/facilities_handler.go:34]: Just noticed this, but you can actually use this to do this directly:
`paginationData := models.NewPaginationInfo(page, perPage, total)`

File: frontend/src/Pages/FacilityManagement.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    // TODO: modify this const
+    const {
+        data: facility,
+        mutate,
+        error,
+        isLoading
+    } = useSWR<ServerResponseMany<Facility>, AxiosError>(`/api/facilities`);
+
+    const facilityData = facility?.data ? facility.data : [];
+    const [, setPageQuery] = useState(1);
```
Comment (calisio) [inline frontend/src/Pages/FacilityManagement.tsx:44]: You will have to use this to modify which page in pagination you are on. See the video:

https://github.com/user-attachments/assets/e8b9b730-786a-4920-8b74-a0c79e0385e0

File: backend/src/handlers/programs_handler_test.go [hunk 4]
```diff
@@ -273,7 +273,7 @@ func getProgramsBySearch(tags string, search string) map[string]any {
 
 func getNewProgramForm() map[string]any {
 	form := make(map[string]any)
-	facilities, err := server.Db.GetAllFacilities()
+	_, facilities, err := server.Db.GetAllFacilities(1, 100)
```
[the remark this answers is not in the document]
Comment (PThorpe92) [reply backend/src/handlers/programs_handler_test.go:276]: i think we can just do it's worth the extra line
 
```go
var facilities []models.Facility
if err := server.Db.Find(&facilities).Error; err != nil {
//...
```

File: backend/src/handlers/facilities_handler.go (hunk 2 above)
Comment (corypride) [reply backend/src/handlers/facilities_handler.go:34]: so we can get rid of the variable used for LastPage altogether, meaning remove line 28 as well?

File: backend/src/handlers/facilites_handler_test.go [hunk 5]
```diff
@@ -185,7 +184,7 @@ func TestHandleDeleteFacility(t *testing.T) {
 		t.Run(test.testName, func(t *testing.T) {
 			var id uint
 			if test.expectedStatusCode == http.StatusNoContent {
-				facility := models.Facility{Name: "Ozark Correctional Center", Timezone: "America/Chicago"}
+				facility := models.Facility{Name: "Ozark Correctional Center", Timezone: "America / Chicago"}
```
Comment (calisio) [inline backend/src/handlers/facilites_handler_test.go:187]: also dont forget to change this timezone without the space also

File: backend/src/handlers/facilities_handler.go (hunk 2 above)
Comment (calisio) [reply backend/src/handlers/facilities_handler.go:34]: Yes, I believe so

File: frontend/src/Pages/FacilityManagement.tsx [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+                        {/* TO DO: this is where SEARCH and SORT will go */}
+                        <div className="flex flex-row gap-x-2"></div>
+                    </div>
+                    <button
+                        className="button"
+                        onClick={() => {
+                            addFacilityModal.current?.showModal();
+                        }}
+                    >
+                        <PlusCircleIcon className="w-4 my-auto" />
```
[the remark this answers is not in the document]
Comment (calisio) [reply frontend/src/Pages/FacilityManagement.tsx:117]: You are right, my bad! Maybe in the future we will update all of them that way its consistent, but for now leave it how you had it 👍

File: backend/src/handlers/auth.go [hunk 7]
```diff
@@ -257,7 +257,8 @@ func (srv *Server) handleResetPassword(w http.ResponseWriter, r *http.Request, l
...
+		if err := srv.Db.UpdateFacility(&facility, 1); err != nil {
```
[the remark this answers is not in the document]
Comment (PThorpe92) [reply backend/src/handlers/auth.go:265]: actually we can just make a quick ticket to go thru the codebase and replace all the instances of the default SuperAdmin + Facility ID's with a constant value. So he's all set to leave this the way it is for now

**Review by calisio — APPROVED** [review]
Good work!

**Pull request merged**